### PR TITLE
feat: add vscode dev container config and improve main `Dockerfile`

### PR DESCRIPTION
**Pull request by akevinge**
Documentation will be added in a separate PR.

### CODE REVIEW

**Comment by akevinge** [comment]
Just for clarification, I got those versions via `pip freeze` and grepping for them. I tested the full stack on Town02 with no issues.

File: docker-compose.yml
Comment (jruths) [inline docker-compose.yml:43]: Do you know of any way for this to fail gracefully (skip it) if someone doesn't have this directory created?

Comment (akevinge) [reply docker-compose.yml:43]: After some thought, I think it'd be best if just didn't use git for version control purposes from within the container (and limit it exclusively to the host). Just too much to configure (SSH credentials, file permissions, etc).

File: docker/entrypoint.sh
Comment (akevinge) [inline docker/entrypoint.sh:34]: Remove `and git settings`

**Comment by akevinge** [comment]
As discussed offline, wait until further testing to approve.

**Review by jpahm — APPROVED** [review]
lgtm assuming the lemon dockerfile issue is the only issue to resolve

**Pull request merged**
